azure/scope/cluster.go

```diff
@@ -295,23 +295,27 @@ func (s *ClusterScope) Subnets() infrav1.Subnets {
 }
 
 // ControlPlaneSubnet returns the cluster control plane subnet.
-func (s *ClusterScope) ControlPlaneSubnet() *infrav1.SubnetSpec {
-	return s.AzureCluster.Spec.NetworkSpec.GetControlPlaneSubnet()
+func (s *ClusterScope) ControlPlaneSubnet() infrav1.SubnetSpec {
+	subnet, _ := s.AzureCluster.Spec.NetworkSpec.GetControlPlaneSubnet()
+	return subnet
 }
 
 // NodeSubnet returns the cluster node subnet.
-func (s *ClusterScope) NodeSubnet() *infrav1.SubnetSpec {
-	return s.AzureCluster.Spec.NetworkSpec.GetNodeSubnet()
+func (s *ClusterScope) NodeSubnet() infrav1.SubnetSpec {
+	subnet, _ := s.AzureCluster.Spec.NetworkSpec.GetNodeSubnet()
+	return subnet
 }
 
 // ControlPlaneRouteTable returns the cluster controlplane routetable.
 func (s *ClusterScope) ControlPlaneRouteTable() *infrav1.RouteTable {
-	return &s.AzureCluster.Spec.NetworkSpec.GetControlPlaneSubnet().RouteTable
+	subnet, _ := s.AzureCluster.Spec.NetworkSpec.GetControlPlaneSubnet()
+	return &subnet.RouteTable
```

Comment:
it might be misleading that this is still returning a pointer but it's not a pointer to the spec value, it's a pointer to that subnet's route table which is itself not a pointer. Maybe it should just return a value, like ControlPlaneSubnet() does.

Reply:
yeah, what you are saying makes sense, will make the func return the spec type without a pointer